ledger/participant-integration-api/src/test/suite/scala/platform/store/backend/UpdateToDbDtoSpec.scala

```diff
@@ -251,47 +253,81 @@ class UpdateToDbDtoSpec extends AnyWordSpec with Matchers {
       )
     }
 
+    val deduplicationPeriods = Table(
+      (
+        "Deduplication period",
+        "Expected deduplication offset",
+        "Expected deduplication time seconds",
+        "Expected deduplication time nanos",
+      ),
+      (None, None, None, None),
+      (
+        Some(DeduplicationOffset(Offset.beforeBegin)),
+        Some(Offset.beforeBegin.toHexString),
+        None,
+        None,
+      ),
+      (
+        Some(DeduplicationDuration(Duration.ofDays(1L))),
+        None,
+        Some(Duration.ofDays(1L).toMinutes * 60L),
+        Some(0),
+      ),
+    )
+
     "handle CommandRejected" in {
-      val completionInfo = state.CompletionInfo(
-        actAs = List(someParty),
-        applicationId = someApplicationId,
-        commandId = someCommandId,
-        optDeduplicationPeriod = None,
-        submissionId = someSubmissionId,
-      )
       val status = StatusProto.of(Status.Code.ABORTED.value(), "test reason", Seq.empty)
-      val update = state.Update.CommandRejected(
-        someRecordTime,
-        completionInfo,
-        new state.Update.CommandRejected.FinalReason(status),
-      )
-      val dtos = UpdateToDbDto(someParticipantId, valueSerialization, compressionStrategy)(
-        someOffset
-      )(update).toList
+      forAll(deduplicationPeriods) {
```

Comment:
I feel like I am hijacking a couple of test cases here but I need to check all the current dedup cases on a rejection and a successful completion (I don't really care which transaction). Let me know if you can think of a better way.

Reply:
Maybe you could have left `"handle TransactionAccepted (single create node)"` alone and added a new test `"handle TransactionAccepted (all deduplication data)"`. This would make it more clear what the purpose of each test is, at the cost of duplicated code.

Reply:
Yes, I think that test purpose clarity is more important so I will do as you suggest before merging.

Reply:
bde3d1e, there's a comprehensive code deduplication work that can be done but I'd rather do it in a separate future PR.